libraries/apollo-gradle-plugin/rules.pro

```diff
@@ -13,10 +13,9 @@
 # For debug
 -keepattributes SourceFile,LineNumberTable
 
-# kotlinpoet uses EnumSetOf that makes a reflexive access to "values"
-# https://github.com/square/kotlinpoet/blob/9952ddcd5095a1fd09c86b9fb07faa347a4c04f0/kotlinpoet/src/main/java/com/squareup/kotlinpoet/PropertySpec.kt#L102
--keepclassmembers class com.squareup.kotlinpoet.KModifier {
-    public static **[] values();
+# Keep all of KotlinPoet to allow usage in Apollo compiler hooks
+-keep class com.squareup.kotlinpoet.** {
```

Comment:
Not sure if it is possible to keep everything, but still relocate it?

Reply:
There's `-dontshrink` but it's global (we already have `-dontobfuscate`. There's a question whether we want to expose a relocated API. It's a bit unexpected for users. 
Maybe we could leverage [IsolationMode](https://docs.gradle.org/current/javadoc/org/gradle/workers/IsolationMode.html) to use kotlinpoet in an isolated classloader?